plugins/woocommerce/includes/class-wc-payment-gateways.php

```diff
@@ -113,6 +118,168 @@ public function init() {
 		}
 
 		ksort( $this->payment_gateways );
+
+		self::add_action( 'wc_payment_gateways_initialized', array( $this, 'on_payment_gateways_initialized' ) );
+		/**
+		 * Hook that is called when the payment gateways have been initialized.
+		 *
+		 * @param WC_Payment_Gateways $wc_payment_gateways The payment gateways instance.
+		 * @since 8.5.0
+		 */
+		do_action( 'wc_payment_gateways_initialized', $this );
+	}
+
+	/**
+	 * Hook into payment gateway settings changes.
+	 *
+	 * @param WC_Payment_Gateways $wc_payment_gateways The WC_Payment_Gateways instance.
+	 * @since 8.5.0
+	 */
+	private function on_payment_gateways_initialized( WC_Payment_Gateways $wc_payment_gateways ) {
+		foreach ( $this->payment_gateways as $gateway ) {
+			$option_key = $gateway->get_option_key();
+			self::add_action(
+				'add_option_' . $option_key,
+				function( $option, $value ) use ( $gateway ) {
+					$this->payment_gateway_settings_option_changed( $gateway, $value, $option );
+				},
+				10,
+				2
+			);
+			self::add_action(
+				'update_option_' . $option_key,
+				function( $old_value, $value, $option ) use ( $gateway ) {
+					$this->payment_gateway_settings_option_changed( $gateway, $value, $option, $old_value );
+				},
+				10,
+				3
+			);
+		}
+	}
+
+	/**
+	 * Callback for when a gateway settings option was added or updated.
+	 *
+	 * @param WC_Payment_Gateway $gateway   The gateway for which the option was added or updated.
+	 * @param mixed              $value     New value.
+	 * @param string             $option    Option name.
+	 * @param mixed              $old_value Old value. `null` when called via add_option_ hook.
+	 * @since 8.5.0
+	 */
+	private function payment_gateway_settings_option_changed( $gateway, $value, $option, $old_value = null ) {
+		if ( ! $this->was_gateway_enabled( $value, $old_value ) ) {
+			return;
+		}
+
+		// This is a change to a payment gateway's settings and it was just enabled. Let's send an email to the admin.
+		// "untitled" shouldn't happen, but just in case.
+		$this->notify_admin_payment_gateway_enabled( $gateway );
+	}
+
+	/**
+	 * Email the site admin when a payment gateway has been enabled.
+	 *
+	 * @param WC_Payment_Gateway $gateway The gateway that was enabled.
+	 * @return bool Whether the email was sent or not.
+	 * @since 8.5.0
+	 */
+	private function notify_admin_payment_gateway_enabled( $gateway ) {
```

Comment:
Minor/nit: we could add the return type (though the method is private, this could still be advantageous if we start making more use of static analyzers and similar).

```suggestion
	private function notify_admin_payment_gateway_enabled( $gateway ): bool {
```

Reply:
I'm a big fan of this, but none of the other methods in that file declare their return type. If we really want to move in that direction I think it should be a more concerted effort.

Reply:
Well, in general, we do not make unnecessary changes to existing code (doing so comes with risk, and increased time investment)—but **do** implement current practices on new code/code we're updating, whenever it's safe and reasonable to do so.

In this way, it feels to me like an analog of how we approach coding standard issues: suppose we have an older class file that's missing docblocks for most of its methods. If we need to add a new method, we definitely also need to add a corresponding docblock—but there is no expectation to go back and address the pre-existing methods.